test/recipes/02-test_errstr.t

```diff
@@ -0,0 +1,41 @@
+#! /usr/bin/env perl
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use OpenSSL::Test;
+use POSIX;
+
+setup('test_errstr');
+
+# We actually have space for up to 4095 error messages,
+# numerically speaking...  but we're currently only using
+# numbers 1 through 127.
+plan tests => 128;
+
+my $i;
+
+for ($i = 1; $i < 128; $i++) {
+    # We know that the system reasons are in library 2
+    my @oerr = run(app([ qw(openssl errstr), sprintf("2%06x", $i) ]),
+                   capture => 1);
+    $oerr[0] =~ s|\R$||;
+    $oerr[0] =~ s|.*system library:||g; # The actual message is last
+
+    my $perr = strerror($i);
+    if ($perr =~ m|Unknown error $i|) {
```

Comment:
Is this a perl thing? Is the perl string "Unknown error" guaranteed to be returned across all platforms and versions?

Reply:
I don't know about guarantee... I know I have observed this over time. I know that's anecdotal, but...

I could check the source...

Reply:
So it seems that in certain environments, we do get strings for unknown errors... And perl has a different one 😕

It also turns out that we get no message on Windows where perl does get something 😕 (I suppose it depends on what libraries the app is linked with) 

Meh...

Reply:
My concern is that this test will fail on obscure platforms/perl versions. Or quite suddenly in the future because the perl maintainers made it "Unknown Error" instead - or some other tweak. Relying on this specific text seems fragile.